### PR TITLE
added decode('utf-8') and 'b'

### PR DESCRIPTION
**Pull request by shirishaganta**
To get rid of str and byte related errors,
changed the script

Signed-off-by: Shirisha Ganta <shiganta@in.ibm.com>

### CODE REVIEW

**Comment by shirishaganta** [comment]
After Patch:
==========
JOB ID     : 15ab3c5642fcf7937fa03adc2fc028c5faed166f
JOB LOG    : /home/tests/results/job-2019-09-30T10.24-15ab3c5/job.log
 (1/1) perf_c2c.py:perf_c2c.test_c2c: PASS (2.88 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 3.77 s
JOB HTML   : /home/tests/results/job-2019-09-30T10.24-15ab3c5/results.html

**Comment by PraveenPenguin** [comment]
@shirishaganta please have other perf test case also be adopted in this commit as it is kind of python3 movement support, i think the same patch needs to be applied other applicable test

File: perf/perf_c2c.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def setUp(self):
 
         # Check for c2c is available in the system.
         output = process.run('perf mem record -e list')
-        if 'ldlat-stores' in output.stderr or 'ldlat-loads' in output.stderr:
+        if b'ldlat-stores' in output.stderr or b'ldlat-loads' in output.stderr:
```
Comment (PraveenPenguin) [inline perf/perf_c2c.py]: decode("utf-8")  can be used as it unique across repo as well this PR

File: perf/perf_uprobe.py [hunk 2]
```diff
@@ -72,10 +72,10 @@ def cmd_verify(self, cmd):
 
     def test_uprobe(self):
         output = self.cmd_verify('%s /usr/bin/perf main' % self.cmdProbe)
-        if 'Added new event' not in output.stderr:
+        if b'Added new event' not in output.stderr:
```
Comment (PraveenPenguin) [inline perf/perf_uprobe.py]: here also `decode("utf-8")` can be used

File: perf/perf_uprobe.py [hunk 3]
```diff
@@ -84,34 +84,34 @@ def test_uprobe(self):
...
-        if 'samples' not in output.stderr:
+        if b'samples' not in output.stderr:
             self.fail("perf: perf.data file not created")
         output = self.cmd_verify(self.report)
 
     def test_uprobe_return(self):
         output = self.cmd_verify('%s ./uprobe_test doit%%return'
                                  % self.cmdProbe)
-        if 'Added new event' not in output.stderr:
+        if b'Added new event' not in output.stderr:
```
Comment (PraveenPenguin) [inline perf/perf_uprobe.py]: here also decode("utf-8") can be used

File: perf/perf_uprobe.py [hunk 4]
```diff
@@ -84,34 +84,34 @@ def test_uprobe(self):
...
             self.fail("perf: probe of uprobe return failed")
         # RHEL
         if self.distro_name == "rhel":
             output = self.cmd_verify('%s__return -- ./uprobe_test'
                                      % self.recProbe)
         else:
             output = self.cmd_verify('%s -aR ./uprobe_test' % self.recProbe)
-        if 'samples' not in output.stderr:
+        if b'samples' not in output.stderr:
```
Comment (PraveenPenguin) [inline perf/perf_uprobe.py]: here also decode("utf-8") can be used

File: perf/perf_uprobe.py [hunk 5]
```diff
@@ -84,34 +84,34 @@ def test_uprobe(self):
...
             self.fail("perf: perf.data file not created")
         output = self.cmd_verify(self.report)
 
     def test_uprobe_variable(self):
         output = self.cmd_verify('%s ./uprobe_test "doit i"' % self.cmdProbe)
-        if 'Added new event' not in output.stderr:
+        if b'Added new event' not in output.stderr:
```
Comment (PraveenPenguin) [inline perf/perf_uprobe.py]: here also decode("utf-8") can be used

File: perf/perf_uprobe.py [hunk 6]
```diff
@@ -84,34 +84,34 @@ def test_uprobe(self):
...
             self.fail("perf: probe of uprobe variable failed")
         if self.distro_name == "rhel":
             output = self.cmd_verify('%s -- ./uprobe_test' % self.recProbe)
         else:
             output = self.cmd_verify('%s -aR ./uprobe_test' % self.recProbe)
-        if 'samples' not in output.stderr:
+        if b'samples' not in output.stderr:
```
Comment (PraveenPenguin) [inline perf/perf_uprobe.py]: here also decode("utf-8") can be used

File: perf/perfmon.py [hunk 7]
```diff
@@ -51,7 +51,7 @@ def test(self):
 
         out = process.system_output('%s ' % os.path.join(
             self.workdir, 'tests/validate'))
-        if 'fail' in out:
+        if b'fail' in out:
```
Comment (PraveenPenguin) [inline perf/perfmon.py]: here also decode("utf-8") can be used

File: perf/perftool.py [hunk 8]
```diff
@@ -73,14 +73,14 @@ def test(self):
         count = 0
         # Built in perf test
         for string in process.run("perf test", ignore_status=True).stderr.splitlines():
-            if 'FAILED' in string:
-                self.log.info("Test case failed is %s" % string)
+            if 'FAILED' in str(string.splitlines()):
```
Comment (PraveenPenguin) [inline perf/perftool.py]: please use decoding instead of type casting in str

File: perf/perftool.py [hunk 9]
```diff
@@ -73,14 +73,14 @@ def test(self):
...
+                self.log.info("Test case failed is %s" % str(string.splitlines()))
```
Comment (PraveenPenguin) [inline perf/perftool.py]: same here

**Comment by shirishaganta** [comment]
@PraveenPenguin I have made all the changes as you requested above

**Pull request merged**